src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,109 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_QUERY =
+            "INSERT INTO books (id, title, price) VALUES (?, ?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_QUERY = "DELETE FROM books WHERE id = ?";
+    private static final String URL = "jdbc:mysql://localhost:3306/book";
+    private static final String LOGIN = "root";
+    private static final String PASSWORD = "root";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = DriverManager.getConnection(URL, LOGIN, PASSWORD)) {
+            PreparedStatement preparedStatement = connection.prepareStatement(CREATE_QUERY);
+            preparedStatement.setObject(1, book.getId());
+            preparedStatement.setString(2, book.getTitle());
+            preparedStatement.setBigDecimal(3, book.getPrice());
+
+            int insertedRows = preparedStatement.executeUpdate();
+            if (insertedRows < 1) {
+                throw new RuntimeException("Expected at least one row to insert");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book  " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = DriverManager.getConnection(URL, LOGIN, PASSWORD)) {
+            PreparedStatement preparedStatement = connection.prepareStatement(FIND_BY_ID_QUERY);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(new Book(id, resultSet.getString("title"),
+                        resultSet.getBigDecimal("price")));
```

Comment:
I would put the creation of the book from ResultSet in a separate private method.

Reply:
Agree